CultivateMVP/JournalOptionsScene/JournalOptionsViewModel.swift

```diff
@@ -1,17 +1,37 @@
 import Foundation
+import RxCocoa
+import RxSwift
 
 protocol JournalOptionsViewModelType {
-    var entries: [Entry] { get }
-    var moodEntry: MoodEntry { get }
+    var viewState: Driver<JournalOptionsViewState> { get }
 }
 
 struct JournalOptionsViewModel: JournalOptionsViewModelType {
 
-    var entries: [Entry]
-    var moodEntry: MoodEntry
+    var viewState: Driver<JournalOptionsViewState> = .never()
+    private var provider: JournalOptionsProvider
+
+    init(journalOptionsProvider: JournalOptionsProvider) {
+        provider = journalOptionsProvider
+            // how to handle an array? Probably include a button, and on tap, go to the next in the array...
+
+        viewState = provider.journalQuestions.map {
+            return JournalOptionsViewState(
+                journalQuestion: $0.first?.question,
+                journalAnswer: "Type",
+                moodQuestion: nil,
+                moodAnswer: nil,
+                nextButtonAction: nil
+                )
+        }.asDriver(onErrorDriveWith: .never())
 
-    init(entries: [Entry], moodEntry: MoodEntry?) {
-        self.entries = entries
-        self.moodEntry = moodEntry ?? MoodEntry(moodQuestion: "Give mood", moodRating: 0)
     }
 }
+
+struct JournalOptionsViewState {
```

Comment:
View state should represent what we can see on the view at once. And consistently define the state. So let's see whats there: 
```swift
struct ViewState {
let date: String
let slider(or whatever it is on the right?)Position: Int
let questionHeader: String
let question: String
let answerType: AnswerType

enum AnswerType { case slider, text }
}
```

Reply:
+ `questionsCount` maybe